### PR TITLE
[llvm] update to v14.0.3 [halide] update to v14.0.0 [mesa] update to 22.0.2

### LINKED ISSUE #23727: [llvm] update to 14
Library name: llvm

New version number: 14

Other information that may be useful (release notes, etc...)
https://github.com/llvm/llvm-project/releases/tag/llvmorg-14.0.0

### PR DESCRIPTION
**Pull request by yurybura**
Updates LLVM to 14.0.3 release.

Fixes #23727

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@yurybura ,This is error link:
https://dev.azure.com/vcpkg/public/_build/results?buildId=70032&view=artifacts&pathAsName=false&type=publishedArtifacts
You can get the entire error log as the picture indicates:
image

**Comment by yurybura** [comment]
@JonLiu1993 Thank you! I will prepare fixes soon. Unfortunately Hilide hasn't released with LLVM 14.0.0 support yet. Seem I have to wait for a while.

File: ports/llvm/0002-fix-install-paths.patch [hunk 1]
```diff
@@ -1,18 +1,19 @@
- clang/cmake/modules/CMakeLists.txt | 4 ++--
```
Comment (Neumann-A) [inline ports/llvm/0002-fix-install-paths.patch:1]: I feel like this patch is unnecessary. `vcpkg_cmake_config_fixup` should move the files instead. This is not a patch which can ever be upstreamed and there is an alternativ solution.

File: ports/llvm/0005-fix-tools-path.patch [hunk 2]
```diff
@@ -2,7 +2,7 @@
  1 file changed, 1 insertion(+), 1 deletion(-)
 
 diff --git a/llvm/tools/llvm-config/llvm-config.cpp b/llvm/tools/llvm-config/llvm-config.cpp
```
Comment (Neumann-A) [inline ports/llvm/0005-fix-tools-path.patch:4]: This patch is invalid and also not upstream able. vcpkg should use `tools/llvm/bin` and duplicate the required stuff from the prefix to make `llvm-config` work normally.

That being said: With the current layout most of the tools cannot be used correctly!

**Comment by ekilmer** [comment]
I have a branch (https://github.com/ekilmer/vcpkg/tree/llvm-14-improvements) that is rebased onto a more recent commit from vcpkg `master` and refactors some of the patching to sort of follow (I think) what @Neumann-A suggested. It also makes source code patches for the other components (clang, lldb, mlir, etc.) to use `LLVM_TOOLS_INSTALL_DIR` more liberally to install their tools at `tools/llvm/bin` (this may or not be the best solution, but it seems to work). See this commit for all of my changes on top of @yurybura's commits https://github.com/ekilmer/vcpkg/commit/9dd7f7f678b4d2b1afe1cfd0e7df4a5a8216ffd4.

Feel free to cherry-pick the commit and/or whatever you need from it to include in this branch @yurybura.

I only tested with `Release` build type on Linux so far (in a custom triplet `x64-linux-rel`, but should also work with the community `x64-linux-release`). I'm building with `x64-linux` now but probably won't be able to test until tomorrow.

I tested the usage of a few projects that link to LLVM libraries and call the `clang` compiler using CMake, so the targets should work, but I haven't tested the other components like linking to Clang libraries or MLIR and others.

Output of `llvm-config` for a few options:
```
$ ./tools/llvm/bin/llvm-config --prefix
/tmp/vcpkg/installed/x64-linux-rel

$ ./tools/llvm/bin/llvm-config --bindir
/tmp/vcpkg/installed/x64-linux-rel/tools/llvm/bin

$ ./tools/llvm/bin/llvm-config --libdir
/tmp/vcpkg/installed/x64-linux-rel/lib

$ ./tools/llvm/bin/llvm-config --cxxflags
-I/tmp/vcpkg/installed/x64-linux-rel/include -std=c++17    -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS

$ ./tools/llvm/bin/llvm-config --libfiles
/tmp/vcpkg/installed/x64-linux-rel/lib/libLLVMWindowsManifest.a /tmp/vcpkg/installed/x64-linux-rel/lib/libLLVMXRay.a /tmp/vcpkg/installed/x64-linux-rel/lib/libLLVMLibDriver.a ...
```

... turns out I missed this one though. It should read `[...]/share/llvm`
```
$ ./tools/llvm/bin/llvm-config --cmakedir
/tmp/vcpkg/installed/x64-linux-rel/lib/cmake/llvm
```

I am also able to use the built sanitizers with the built `clang` compiler (located at `/tmp/vcpkg/installed/x64-linux-rel/tools/llvm/lib/clang/14.0.0/lib/linux/libclang_rt.asan-x86_64.a`):
```bash
/tmp $ ./vcpkg/installed/x64-linux-rel/tools/llvm/bin/clang -fsanitize=address -g -v leak.c
clang version 14.0.0 (git@github.com:ekilmer/vcpkg.git 6a2a0cbbd022226ca57b1ebd57cb6b2c03d773c0)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /tmp/./vcpkg/installed/x64-linux-rel/tools/llvm/bin
Found candidate GCC installation: /usr/lib/gcc/i686-redhat-linux/11
Found candidate GCC installation: /usr/lib/gcc/x86_64-redhat-linux/11
Selected GCC installation: /usr/lib/gcc/x86_64-redhat-linux/11
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Selected multilib: .;@m64
 "/tmp/vcpkg/installed/x64-linux-rel/tools/llvm/bin/clang-14" -cc1 -triple x86_64-unknown-linux-gnu -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name leak.c -mrelocation-model static -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debug-info-kind=constructor -dwarf-version=5 -debugger-tuning=gdb -v -fcoverage-compilation-dir=/tmp -resource-dir /tmp/vcpkg/installed/x64-linux-rel/tools/llvm/lib/clang/14.0.0 -internal-isystem /tmp/vcpkg/installed/x64-linux-rel/tools/llvm/lib/clang/14.0.0/include -internal-isystem /usr/local/include -internal-isystem /usr/lib/gcc/x86_64-redhat-linux/11/../../../../x86_64-redhat-linux/include -internal-externc-isystem /include -internal-externc-isystem /usr/include -fdebug-compilation-dir=/tmp -ferror-limit 19 -fsanitize=address -fsanitize-system-ignorelist=/tmp/vcpkg/installed/x64-linux-rel/tools/llvm/lib/clang/14.0.0/share/asan_ignorelist.txt -fsanitize-address-use-after-scope -fno-assume-sane-operator-new -fgnuc-version=4.2.1 -fcolor-diagnostics -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /tmp/leak-a3cf52.o -x c leak.c
clang -cc1 version 14.0.0 based upon LLVM 14.0.0 default target x86_64-unknown-linux-gnu
ignoring nonexistent directory "/usr/lib/gcc/x86_64-redhat-linux/11/../../../../x86_64-redhat-linux/include"
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /tmp/vcpkg/installed/x64-linux-rel/tools/llvm/lib/clang/14.0.0/include
 /usr/local/include
 /usr/include
End of search list.
 "/usr/bin/ld" --hash-style=gnu --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o a.out /usr/lib/gcc/x86_64-redhat-linux/11/../../../../lib64/crt1.o /usr/lib/gcc/x86_64-redhat-linux/11/../../../../lib64/crti.o /usr/lib/gcc/x86_64-redhat-linux/11/crtbegin.o -L/usr/lib/gcc/x86_64-redhat-linux/11 -L/usr/lib/gcc/x86_64-redhat-linux/11/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/tmp/vcpkg/installed/x64-linux-rel/tools/llvm/bin/../lib -L/lib -L/usr/lib --whole-archive /tmp/vcpkg/installed/x64-linux-rel/tools/llvm/lib/clang/14.0.0/lib/linux/libclang_rt.asan_static-x86_64.a --no-whole-archive --whole-archive /tmp/vcpkg/installed/x64-linux-rel/tools/llvm/lib/clang/14.0.0/lib/linux/libclang_rt.asan-x86_64.a --no-whole-archive --dynamic-list=/tmp/vcpkg/installed/x64-linux-rel/tools/llvm/lib/clang/14.0.0/lib/linux/libclang_rt.asan-x86_64.a.syms /tmp/leak-a3cf52.o --no-as-needed -lpthread -lrt -lm -ldl -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed /usr/lib/gcc/x86_64-redhat-linux/11/crtend.o /usr/lib/gcc/x86_64-redhat-linux/11/../../../../lib64/crtn.o

/tmp $ ./a.out
string is: Hello world!

=================================================================
==3455735==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 100 byte(s) in 1 object(s) allocated from:
    #0 0x49f2fe in __interceptor_malloc /tmp/vcpkg/buildtrees/llvm/src/org-14.0.0-8914b0de2d.clean/compiler-rt/lib/asan/asan_malloc_linux.cpp:69:3
    #1 0x4da9df in main /tmp/leak.c:6:15
    #2 0x7fc76639955f in __libc_start_call_main (/lib64/libc.so.6+0x2d55f) (BuildId: 5d63c4327ba6cdd1e8b99eb1c0a9f056a98c67e1)

SUMMARY: AddressSanitizer: 100 byte(s) leaked in 1 allocation(s).
```

File: ports/llvm/0002-fix-install-paths.patch (hunk 1 above)
Comment (yurybura) [reply ports/llvm/0002-fix-install-paths.patch:1]: > I feel like this patch is unnecessary. `vcpkg_cmake_config_fixup` should move the files instead. This is not a patch which can ever be upstreamed and there is an alternativ solution.

CMake paths are used inside llvm-config utility tool. Without this patch I should fix C++ code of llvm-config (`./llvm-config --cmakedir` will return incorrect path) and modify [name]Config.cmake after `vcpkg_cmake_config_fixup` (see https://github.com/ekilmer/vcpkg/commit/9dd7f7f678b4d2b1afe1cfd0e7df4a5a8216ffd4#diff-fc93cddf7017459f44baa59103106dd40a4482ddbd93afd4772cc265c18f9332R282-R285).
I think the simplest way to fix original CMake.

File: ports/llvm/0005-fix-tools-path.patch (hunk 2 above)
Comment (yurybura) [reply ports/llvm/0005-fix-tools-path.patch:4]: > This patch is invalid and also not upstream able. vcpkg should use `tools/llvm/bin` and duplicate the required stuff from the prefix to make `llvm-config` work normally.
> 
> That being said: With the current layout most of the tools cannot be used correctly!

Default path to the tools in VCPKG documentation is `tools/${PORT}` (see [vcpkg_copy_tools.md](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_copy_tools.md)). Could you explain why it should be `tools/${PORT}/bin`?

Comment (Neumann-A) [reply ports/llvm/0005-fix-tools-path.patch:4]: see https://github.com/microsoft/vcpkg/pull/23734#issuecomment-1094068078
in #23987 I also use that layout to make the tools work.
Simply the tools expect the `prefix` to be at `bin/..`. So that the libs are located at `prefix/lib` and the includes being located at `prefix/include`. The tools expect a fixed layout to add standard compiler search paths for libraries and includes. It took me two days to debug the `pgmath` build and why it didn't work using the current llvm layout. Turns out clang-cl will use the default cl headers insteads of its internal headers for intrinsics if the layout is incorrect because the header search path was incorrect.  

As such you want to have compiler internal headers somewhere at `/tools/llvm/include` instead of `/include`. The same goes for internally libraries which are not needed by anyone else -> `/tools/llvm/lib`

Basically not all installed libs/headers are intended for downstream consumption.

**Comment by JonLiu1993** [comment]
@yurybura ,Thanks for your contribution, I tested the features with x64-windows triplet on my local machine but failed:
````
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: ninja -v
    Working Directory: F:/Feature-test/llvm/vcpkg/buildtrees/llvm/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      F:\Feature-test\llvm\vcpkg\buildtrees\llvm\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:346 (vcpkg_execute_required_process)
  ports/llvm/portfile.cmake:242 (vcpkg_cmake_configure)
  scripts/ports.cmake:145 (include)


error: building llvm:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
````
This is error log:
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/8530356/config-x64-windows-out.log)

**Comment by yurybura** [comment]
@JonLiu1993 Please add your install command: vcpkg install llvm[???]:x64-windows

**Comment by JonLiu1993** [comment]
> @JonLiu1993 Please add your install command: vcpkg install llvm[???]:x64-windows

 ./vcpkg install llvm[*]:x64-windows

**Comment by yurybura** [comment]
> ./vcpkg install llvm[*]:x64-windows
@JonLiu1993 This test won't work due to some options are alternative.

**Comment by JonLiu1993** [comment]
> > ./vcpkg install llvm[*]:x64-windows
> 
> @JonLiu1993 This test won't work due to some options are not supported with MSVC.
> 
> I've just tested `./vcpkg install llvm[bolt,clang,clang-tools-extra,compiler-rt,libclc,lld,lldb,mlir,polly,flang,target-all,enable-ffi,enable-libxml2,tools,utils]:x64-windows-release`. CMake configuration done.

When I test in the local windows environment, I will block the unsupported features and use the ./vcpkg install llvm[*]:x64-windows command to test. Will this work?

**Review by JonLiu1993 — APPROVED** [review]

File: ports/llvm/0005-fix-tools-path.patch (hunk 2 above)
Comment (BillyONeal) [reply ports/llvm/0005-fix-tools-path.patch:4]: I agree with @Neumann-A that this is a problem, but I don't think it's a problem introduced by this version update, so I don't think we should hold the version update over it.

It's unfortunate that LLVM tries its best to be a pain to consume from things that aren't part of the LLVM project itself.

**Comment by BillyONeal** [comment]
> I have a branch (https://github.com/ekilmer/vcpkg/tree/llvm-14-improvements) that is rebased onto a more recent commit from vcpkg `master` and refactors some of the patching to sort of follow (I think) what @Neumann-A suggested.

Are you interested in or have you made that its own PR? It would be good to make a version update and functionality change like that as separate PRs.

File: ports/mesa/clang.patch [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+   pre_args += '-D_ISOC11_SOURCE'
+diff --git a/src/compiler/nir/nir_lower_atomics_to_ssbo.c b/src/compiler/nir/nir_lower_atomics_to_ssbo.c
+index c1799d0b9d0af70148928a4b224172c188c64194..448f63bdc7cc3b9cae123e3c0abe79b4b5af380b 100644
+--- a/src/compiler/nir/nir_lower_atomics_to_ssbo.c
++++ b/src/compiler/nir/nir_lower_atomics_to_ssbo.c
+@@ -27,10 +27,6 @@
+ #include "nir.h"
+ #include "nir_builder.h"
+ 
+-#if defined(_WIN32) && !defined(snprintf)
```
Comment (BillyONeal) [inline ports/mesa/clang.patch:33]: Do you think upstream would accept a change to

`#if defined(_WIN32) && !defined(snprintf) && !defined(_UCRT)`

or something like that as a perma-fix?

File: ports/mesa/clang.patch [hunk 4]
```diff
@@ -0,0 +1,52 @@
+diff --git a/meson.build b/meson.build
```
Comment (BillyONeal) [inline ports/mesa/clang.patch:1]: Has this change been submitted upstream?

File: ports/mesa/clover.patch [hunk 5]
```diff
@@ -0,0 +1,63 @@
+From b7c465dc43693cb1ad7e073cc7d8f89f5cfbd0ac Mon Sep 17 00:00:00 2001
+From: pal1000 <liviuprodea@yahoo.com>
+Date: Sat, 13 Nov 2021 11:39:58 +0200
+Subject: [PATCH] clover: Use static pipe loader on Windows
```
Comment (BillyONeal) [inline ports/mesa/clover.patch:4]: This looks like an upstream submission but https://github.com/mesa3d/mesa/commit/b7c465dc43693cb1ad7e073cc7d8f89f5cfbd0ac 404s. Can you indicate where upstream accepted this?

File: ports/mesa/def-fixes.patch [hunk 6]
```diff
@@ -0,0 +1,3419 @@
+diff --git a/bin/gen_vs_module_defs.py b/bin/gen_vs_module_defs.py
+new file mode 100644
+index 0000000000000000000000000000000000000000..dff8ab36f92ffbc18dbb2779563c25df567bb7b0
+--- /dev/null
++++ b/bin/gen_vs_module_defs.py
+@@ -0,0 +1,98 @@
++#!/usr/bin/env python3
++# Copyright Â© 2021-2021 Yonggang Luo
```
Comment (BillyONeal) [inline ports/mesa/def-fixes.patch:8]: Is this you? If not, where did it come from? How will upstream see something like this?

It seems like a new file, should it just be in the port directory and `file(COPY`'d in rather than in a patch file like this?

File: ports/mesa/def-fixes.patch [hunk 7]
```diff
@@ -0,0 +1,3419 @@
...
++++ /dev/null
+@@ -1,4 +0,0 @@
+-EXPORTS
+-vk_icdNegotiateLoaderICDInterfaceVersion
+-vk_icdGetInstanceProcAddr
+-vk_icdGetPhysicalDeviceProcAddr
+diff --git a/src/egl/main/egl.def b/src/egl/main/egl.def
+deleted file mode 100644
+index 54ae76b330929db865dc4ef0fd1723a88dc47979..0000000000000000000000000000000000000000
+--- a/src/egl/main/egl.def
```
Comment (BillyONeal) [inline ports/mesa/def-fixes.patch:147]: Ditto can this be done with `file(RENAME` or similar to avoid needing to copy so much of the upstream into the catalog?

File: ports/mesa/portfile.cmake [hunk 8]
```diff
@@ -3,16 +3,12 @@
...
-    swravx512-post-static-link.patch
-    # Fix swr build with MSVC
-    swr-msvc-2.patch
-    # Fix swr build with LLVM 13
-    swr-llvm13.patch
-    # Fix radv MSVC build with LLVM 13
-    radv-msvc-llvm13-2.patch
-    # Fix d3d10sw MSVC build
-    d3d10sw.patch
+    # Fix symbols exporting for MinGW GCC x86
```
Comment (BillyONeal) [inline ports/mesa/portfile.cmake:6]: If this huge patch is only for MinGW I'm even more squeamish about it. Upstream should be involved here.

File: ports/mesa/portfile.cmake [hunk 9]
```diff
@@ -3,16 +3,12 @@
 
 # Patches are from https://github.com/pal1000/mesa-dist-win/tree/master/patches
```
Comment (BillyONeal) [inline ports/mesa/portfile.cmake:4]: Can we actually download them?

**Review by BillyONeal — CHANGES_REQUESTED** [review]

**Comment by BillyONeal** [comment]
I'm only waiting for clarification that upstream knows about those huge patches. Everything in LLVM proper looks good.

**Comment by ekilmer** [comment]
> > I have a branch (https://github.com/ekilmer/vcpkg/tree/llvm-14-improvements) that is rebased onto a more recent commit from vcpkg `master` and refactors some of the patching to sort of follow (I think) what Neumann-A suggested.
> 
> Are you interested in or have you made that its own PR? It would be good to make a version update and functionality change like that as separate PRs.

I have not yet made that into its own PR. If that were to happen, would it be PR'd onto yurybura's branch, assuming it hasn't been merged yet, or vcpkg's `master` branch with mention of this PR?

I don't want to step on yurybura's great work in this PR or distract from other discussions with the major version upgrade, so I was thinking about waiting until this PR was merged before opening another one. I'm happy to do the option you think is best.

**Comment by Neumann-A** [comment]
Please re-evalute the openmp build on windows. I fear the comment in the portfile is out of date considering that visual studio ships openmp from llvm. Also do we need to take care about: `LIBOMP_ARCH` ?

**Comment by Neumann-A** [comment]
Another question would be if the LLVM build should be split up into modular builds in the future. Stuff like openmp does not need to be build with the main LLVM project.

File: ports/mesa/clang.patch (hunk 3 above)
Comment (yurybura) [reply ports/mesa/clang.patch:33]: This is not my patch. I copied it from https://github.com/pal1000/mesa-dist-win/tree/master/patches

File: ports/mesa/portfile.cmake (hunk 9 above)
Comment (yurybura) [reply ports/mesa/portfile.cmake:4]: I'm not sure that we can use these patches without testing. Better to copy them to the port.

**Comment by yurybura** [comment]
> I'm only waiting for clarification that upstream knows about those huge patches. Everything in LLVM proper looks good.

@BillyONeal 
I copied the mesa related patches from https://github.com/pal1000/mesa-dist-win/tree/master/patches and then tested the build. I don't know much about the Mesa project, but without fixing this port, I can't update LLVM... I have to make CI happy)

**Comment by JonLiu1993** [comment]
@BillyONeal, Could you please help take a review?

**Comment by BillyONeal** [comment]
> @BillyONeal I copied the mesa related patches from https://github.com/pal1000/mesa-dist-win/tree/master/patches and then tested the build. I don't know much about the Mesa project, but without fixing this port, I can't update LLVM... I have to make CI happy)

Then I think we need to confirm that upstream of Mesa has had an opportunity to look at those patches before updating LLVM. (We can't ignore our policy of respecting the actual owners of the things we purport to ship just because there's a shared dependency we want to update)

File: ports/llvm/portfile.cmake [hunk 10]
```diff
@@ -180,6 +158,23 @@ if("pstl" IN_LIST FEATURES)
     list(APPEND LLVM_ENABLE_PROJECTS "pstl")
 endif()
 
+set(LLVM_ENABLE_RUNTIMES)
+if("libcxx" IN_LIST FEATURES)
+    if(VCPKG_TARGET_IS_WINDOWS)
+        message(FATAL_ERROR "Building libcxx with MSVC is not supported, as cl doesn't support the #include_next extension.")
+    endif()
```
Comment (Neumann-A) [inline ports/llvm/portfile.cmake:165]: Building with `clang-cl` works? If yes this needs a compiler check not a target check. Compiler check -> call `vcpkg_get_cmake_vars`

File: ports/llvm/portfile.cmake [hunk 11]
```diff
@@ -180,6 +158,23 @@ if("pstl" IN_LIST FEATURES)
...
+    list(APPEND LLVM_ENABLE_RUNTIMES "libcxx")
+endif()
+if("libcxxabi" IN_LIST FEATURES)
+    if(VCPKG_TARGET_IS_WINDOWS)
+        message(FATAL_ERROR "Building libcxxabi with MSVC is not supported. Disable it until issues are fixed.")
+    endif()
```
Comment (Neumann-A) [inline ports/llvm/portfile.cmake:171]: Same as above

File: ports/llvm/portfile.cmake (hunk 10 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake:165]: > z_vcpkg_get_cmake_vars
> **Only for internal use in vcpkg helpers. Behavior and arguments will change without notice.**
> Runs a cmake configure with a dummy project to extract certain cmake variables

@Neumann-A Maybe you know another way?

Comment (BillyONeal) [reply ports/llvm/portfile.cmake:165]: You want `vcpkg_cmake_get_vars` from https://github.com/microsoft/vcpkg/blob/master/ports/vcpkg-cmake/vcpkg_cmake_get_vars.cmake

Comment (Neumann-A) [reply ports/llvm/portfile.cmake:165]:
```
    vcpkg_cmake_get_vars(cmake_vars_file)
    include("${cmake_vars_file}")
    if(VCPKG_DETECTED_CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
```

File: ports/llvm/portfile.cmake (hunk 11 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: Fixed.

File: ports/llvm/portfile.cmake (hunk 10 above)
Comment (yurybura) [reply ports/llvm/portfile.cmake]: Fixed.

File: ports/mesa/clover.patch (hunk 5 above)
Comment (yurybura) [reply ports/mesa/clover.patch:4]: This is not my patch. I copied it from https://github.com/pal1000/mesa-dist-win/tree/master/patches

File: ports/mesa/portfile.cmake (hunk 8 above)
Comment (yurybura) [reply ports/mesa/portfile.cmake:6]: I can remove it if it isn't acceptable.

File: ports/mesa/clang.patch (hunk 4 above)
Comment (yurybura) [reply ports/mesa/clang.patch:1]: This is not my patch. I copied it from https://github.com/pal1000/mesa-dist-win/tree/master/patches

File: ports/mesa/portfile.cmake (hunk 8 above)
Comment (yurybura) [reply ports/mesa/portfile.cmake:6]: This is not my patch. I copied it from https://github.com/pal1000/mesa-dist-win/tree/master/patches

**Comment by BillyONeal** [comment]
Thanks for burninating a lot of patches, and updating LLVM!
